File: output/deepseek_r1/avatar/direct_translation/java/python/successful_translation/codeforces_274_A.py

import sys

data = list(map(int, sys.stdin.read().split()))
n, k = data[0], data[1]
arr = data[2:2 + n]
arr.sort()
s = set()
count = 0
for num in arr:
    if num % k != 0:
        count += 1
        s.add(num)
    else:
        div = num // k
        if div not in s:
            count += 1
            s.add(num)
print(count)

File: output/deepseek_r1/avatar/direct_translation/python/java/successful_translation/codeforces_274_A.java

import java.util.*;

public class codeforces_274_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();
        int k = scanner.nextInt();
        int[] arr = new int[num];
        for (int i = 0; i < num; i++) {
            arr[i] = scanner.nextInt();
        }
        Arrays.sort(arr);
        Set<Integer> set = new HashSet<>();
        int max = 0;
        for (int x : arr) {
            if (x % k != 0 || !set.contains(x / k)) {
                set.add(x);
            }
            max = Math.max(max, set.size());
        }
        System.out.println(max);
    }
}

